levels/levels.go

```diff
@@ -68,7 +68,17 @@ func (c *Challenge) AddLevel(level Level) {
 
 func (c *Challenge) CheckCurrentLevel() bool {
 	level := c.IDToIndex(*c.CurrentLevel)
-	return CmdOK(c.Levels[level].TestCmd)
+	passed, next_level := CmdOK(c.Levels[level].TestCmd)
+	if next_level == "" {
+		return passed
+	} else {
+		index := c.LevelNameToIndex(next_level)
+		if index == -1 {
+			return false
+		}
+		*c.CurrentLevel = IndexToID(index, c.Name)
```

Comment:
Wouldn't this cause you to jump one level further? You set level to X and then (I guess) it gets incremented by one if the result of this is true.